Converts ghosts mobs to a basic mob, slightly changes how ghosts are given their identities

Rhials opened the pull request:
## About The Pull Request

Performs a very simple conversion of ghosts to basic mobs. They're relatively lacking in unique behavior in their current state, but they didn't really have any when they were simple animals either. They retaliate, they float, they do everything they did before, but now as a basic mob.

Kilostation and Derelict Station 6 have had their mapped-in ghosts converted to the new type as well. I'm unsure if there are any other mapped instances of ghosts that will need changing, but please let me know if there are.

Also in this PR is a slight rework of how ghost's get their hair/name. As an ADDED BONUS, ghosts now get random hair to gow ith their randomly generated names, rather than hair only being only possible through varediting. (Thank you tattle for helping me figure this out).
## Why It's Good For The Game

Strikes another simple_animal from existence. Gives ghost mobs a bit more _visual flair_.
## Changelog
:cl:
code: Ghosts are now basic mobs! You might not notice any difference, but...
code: Ghost mobs now spawn with hair and sometimes facial hair.
/:cl:

Rhials committed:
first pass

makes ghosts a really shitty really basic retaliate AI until I can start making better thigns

also does autodocs :)

Rhials committed:
finishes AI, removes the freeze thing

apparently simplemob ghosts are literally never referenced outside of maps, meaning DM compiled without me having to change anything. Lol

Rhials committed:
git commit

Rhials committed:
folder change

Rhials committed:
randomized identities!!

ghosts will now have custom hair to go with their custom name. Also lets you opt out of random identity stuff and just make a normal ghost

Rhials committed:
fixes kilo

Rhials committed:
fixes derelict 6

Rhials committed:
hair works thank you tattle :D

Rhials committed:
thants not supposed to be there

Rhials committed:
comment stff

Rhials committed:
big oopsies with the name

Rhials committed:
puts the freeze thing back

looks like I wasnt supposed to remove this OOPS

Rhials committed:
and this also oops

Rhials committed:
better comments, better give_identity

Mothblocks reviewed (changes requested):
needs an update paths script

Rhials committed:
updatepaths

dear diary

today i learned about the tools folder and about scripting

im going to look into this further and hopefully do some cool things :D

Rhials committed:
death loots list statico

Rhials committed:
other two things

san7890 commented:
```txt
code/modules/unit_tests/simple_animal_freeze.dm:288:error: /mob/living/simple_animal/hostile/retaliate/ghost: undefined type path
```

Your code isn't compiling for the Unit Tests scope because you need to remove the old path from the big-ass list in the Simple Animal Freeze file.

Rhials committed:
get outta here

Rhials committed:
Merge remote-tracking branch 'upstream/master' into basic-mob-hell

Rhials committed:
aaaand you

jlsnow301 committed:
Merge branch 'master' into basic-mob-hell
